data_science/01.00-addition-laws.py

```diff
@@ -0,0 +1,416 @@
+    return
+
+
+@app.cell
+def _(mo):
+    # Interactive die probability calculator
+    die_events = mo.ui.multiselect(
+        options=["1", "2", "3", "4", "5", "6"],
+        value=["1", "2"],
+        label="Select die numbers"
+    )
+    return (die_events,)
+
+
+@app.cell
+def _(die_events, mo, plt):
+    selected = [int(x) for x in die_events.value]
+    die_prob = len(selected) / 6
+
+    # Create simple bar plot
+    plt.figure(figsize=(8, 4))
+    plt.bar(['Selected', 'Others'], 
+            [die_prob, 1-die_prob],
+            color=['#2ecc71', '#e74c3c'])
+    plt.title('Probability Distribution')
+    plt.ylabel('Probability')
+    plt.ylim(0, 1)
+
+    die_result = mo.md(f"""
+    ### Probability Analysis
+
+    Selected numbers: {', '.join(die_events.value)}
+
+    P(selected) = {die_prob:.3f}
+
+    This is a simple addition because each number can only come up once!
+    """)
+
+    mo.hstack([plt.gcf(), die_result])
+    return die_prob, die_result, selected
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        ## 🚫 Mutually Exclusive Events
+
+        Events are mutually exclusive when they cannot happen at the same time.
+
+        For mutually exclusive events A and B:
+
+        $P(A \cup B) = P(A) + P(B)$
+
+        Let's visualize this with a weather example!
+        """
+    )
+    return
+
+
+@app.cell
+def _(mo, rainy_prob, sunny_prob):
+    mo.hstack([sunny_prob, rainy_prob])
+    return
+
+
+@app.cell
+def _(mo):
+    # Weather probability selector
+    sunny_prob = mo.ui.slider(
+        value=0.3,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Sunny)"
+    )
+    rainy_prob = mo.ui.slider(
+        value=0.4,
+        start=0,
+        stop=1,
+        step=0.1,
+        label="P(Rainy)"
+    )
+    return rainy_prob, sunny_prob
+
+
+@app.cell
+def _(mo, plt, rainy_prob, sunny_prob, venn2):
+    weather_total = sunny_prob.value + rainy_prob.value
+
+    # Create Venn diagram
+    plt.figure(figsize=(8, 5))
+    v = venn2(subsets=(sunny_prob.value, rainy_prob.value, 0),
+              set_labels=('Sunny', 'Rainy'))
+
+    # Customize colors
+    if v:
+        v.get_patch_by_id('10').set_color('#f1c40f')
+        v.get_patch_by_id('01').set_color('#3498db')
+
+    plt.title('Mutually Exclusive Weather Events')
+
+    weather_result = mo.md(f"""
+    ### Weather Probability Analysis
+
+    P(Sunny) = {sunny_prob.value}
+    P(Rainy) = {rainy_prob.value}
+    P(Sunny or Rainy) = {weather_total}
+
+    Status: {"✅ Valid" if weather_total <= 1 else "❌ Invalid - Total exceeds 1!"}
+    """)
+
+    mo.hstack([plt.gcf(), weather_result])
+    return v, weather_result, weather_total
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        ## 🤝 General Addition Rule
+
+        When events can overlap, we need to subtract the overlap:
+
+        $P(A \cup B) = P(A) + P(B) - P(A \cap B)$
+
+        Let's explore this with a student club example!
```

Comment:
Before going into an example, it is worth giving an English language description of what the math says.

"This equation means the probability of A or B both happening is the sum of the probabilities of A happening and B happening, minus the probability of both of them happening at the same time."

Then, give an example in English before jumping into an interactive example.